src/visual.ts

```diff
@@ -796,16 +867,24 @@ module powerbi.extensibility.visual {
 
         private static updateValueOfNode(node: SankeyDiagramNode): void {
             node.inputWeight = node.links.reduce((previousValue: number, currentValue: SankeyDiagramLink) => {
-                return previousValue + (currentValue.destination === node
-                    ? currentValue.weigth
+                return previousValue + (currentValue.destination === node && currentValue.destination !== currentValue.source
```

Comment:
Not sure if it might affect the performance but we should check if we could merged these reducers in one forEach in order not to enumerate the same collection four times.